### ISSUE TITLE
an MPI error occurs when testing the "heatTransfer write" example using MPI aggregate method

### ISSUE THREAD

**Issue by lwan86** [ornladios/ADIOS2#835]
This is MPI error:
"*** The MPI_Comm_free() function was called after MPI_FINALIZE was invoked.
*** This is disallowed by the MPI standard."

It seems that "MPI_Comm_free(&m_Comm)" function is called in the destructor of "MPIAggregator" class, which is called after the "MPI_Finalize()" at the end of the main function of the "heatTransfer write" example.

**Comment by williamfgc** [comment]
yeah, it's not being called. Which system and mpi implementation are you using?

**Comment by lwan86** [comment]
I tested it on both my mac laptop and ubuntu vm. The MPI implementation I'm using is Open MPI (version 1.10.2 on my mac and 2.1.1 on the vm).

**Comment by williamfgc** [comment]
@lwan86 can you verify that the current master works after rebasing onto your branch?

**Comment by lwan86** [comment]
@williamfgc I just tested the current master, the same error still occurs.  Here is the configuration I used.

<?xml version="1.0"?>

<adios-config>

    <io name="writer">
        <engine type="BPFile">

            <parameter key="Threads" value="1"/>

            <parameter key="ProfileUnits" value="Microseconds"/>

            <parameter key="InitialBufferSize" value="16Kb"/> 

            <parameter key="FlushStepsCount" value="1"/>

	    <parameter key="Substreams" value="2"/> 
            
        </engine>

        <transport type="File">

            <parameter key="Library" value="POSIX"/>
            
            <parameter key="ProfileUnits" value="Microseconds"/>
            
        </transport>
        
    </io>
</adios-config>

**Comment by williamfgc** [comment]
My bad, that was for N-to-N. Now the aggregator comm is actually freed. Good catch! The current master should work.
